src/Auth.php

```diff
@@ -19,30 +19,35 @@ class Auth {
     private $tokenDetails;
     private $ably;
 
-    public function __construct(AblyRest $ably, ClientOptions $options) {
+    public function __construct( AblyRest $ably, ClientOptions $options ) {
         $this->authOptions = new AuthOptions($options);
         $this->ably = $ably;
 
-        if ($this->authOptions->key && empty($this->authOptions->clientId)) {
+        if ( $this->authOptions->key && empty( $this->authOptions->clientId ) ) {
             $this->basicAuth = true;
-            Log::d('Auth: anonymous, using basic auth');
+            Log::d( 'Auth: anonymous, using basic auth' );
+
+            if ( !$options->tls ) {
+                log::e( 'Trying to use basic key auth over unsecure connection' );
```

Comment:
Ok, how about simply changing to "Using an API key over an insecure connection with basic auth is prohibited"